src/Servers/Kestrel/Transport.Libuv/src/Internal/LibuvConnection.cs

```diff
@@ -123,19 +161,25 @@ public async Task Start()
         public override void Abort(ConnectionAbortedException abortReason)
         {
             _abortReason = abortReason;
-            
+
             // Cancel WriteOutputAsync loop after setting _abortReason.
             Output.CancelPendingRead();
 
             // This cancels any pending I/O.
             Thread.Post(s => s.Dispose(), _socket);
         }
 
-        // Only called after connection middleware is complete which means the ConnectionClosed token has fired.
-        public void Dispose()
+        public override async ValueTask DisposeAsync()
         {
+            Transport.Input.Complete();
+            Transport.Output.Complete();
+
+            if (_task != null)
+            {
+                await _task;
+            }
+
             _connectionClosedTokenSource.Dispose();
-            _connectionClosingCts.Dispose();
```

Comment:
Why did this get removed?

Reply:
The transport no longer owns this but it's not being disposed right now. It's in KestrelConnection.

Reply:
Fixed